web/src/message_fetch.js

```diff
@@ -45,11 +45,17 @@ const consts = {
 
 function process_result(data, opts) {
     let messages = data.messages;
-
-    messages = messages.map((message) => message_helper.process_new_message(message));
     const has_found_oldest = opts.msg_list?.data.fetch_status.has_found_oldest() ?? false;
     const has_found_newest = opts.msg_list?.data.fetch_status.has_found_newest() ?? false;
 
+    const filter = opts.msg_list ? opts.msg_list.data.filter : opts.msg_list_data.filter;
+    const is_data_reliable =
```

Comment:
In this context, maybe call this `is_data_reliable_for_stream_history`? In the context of the main `message_fetch` code path, this name doesn't make any sense, of course the messages themselves are reliable data.